OpenRA.Mods.Common/Projectiles/GravityBomb.cs

```diff
@@ -47,17 +48,18 @@ public class GravityBomb : IProjectile, ISync
 		readonly GravityBombInfo info;
 		readonly Animation anim;
 		readonly ProjectileArgs args;
+		readonly WVec acceleration;
 		[Sync] WVec velocity;
 		[Sync] WPos pos;
-		[Sync] WVec acceleration;
 
 		public GravityBomb(GravityBombInfo info, ProjectileArgs args)
 		{
 			this.info = info;
 			this.args = args;
 			pos = args.Source;
-			velocity = new WVec(WDist.Zero, WDist.Zero, -info.Speed);
-			acceleration = new WVec(WDist.Zero, WDist.Zero, info.Acceleration);
+			var facing = (args.PassiveTarget - pos).Yaw.Facing;
+			velocity = -info.Velocity.Rotate(WRot.FromFacing(facing));
```

Comment:
In fact, (nearly?) all of our projectile types rely on `SourceActor` not being null in one way or another.  Multi-stage projectiles aren't officially a thing, and I expect will need some plumbing rework to be properly supported.